### PR TITLE
Add support for multiple General Considerations

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1634

### Change description ###
Move general referral to general referrals list on legal-advisor-general-consideration
Blank general referral fields after copying to general referrals list

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralReferral.java [hunk 1]
```diff
@@ -95,4 +96,22 @@
...
+            .generalApplicationAddedDate(generalApplicationAddedDate)
+            .generalReferralType(generalReferralType)
+            .alternativeServiceMedium(alternativeServiceMedium)
+            .generalReferralJudgeOrLegalAdvisorDetails(generalReferralJudgeOrLegalAdvisorDetails)
+            .generalReferralFeeRequired(generalReferralFeeRequired)
+            .generalReferralFee(generalReferralFee)
+            .generalReferralDecision(generalReferralDecision)
+            .generalReferralDecisionDate(generalReferralDecisionDate)
+            .generalReferralDecisionReason(generalReferralDecisionReason)
+            .build();
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralReferral.java:115]: can we not use `@Builder(toBuilder=true)` on GeneralReferral and get a copy of it ?

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java [hunk 2]
```diff
@@ -110,6 +110,13 @@
     @Builder.Default
     private GeneralReferral generalReferral = new GeneralReferral();
 
+    @CCD(
+        label = "General Referrals",
+        typeOverride = Collection,
+        typeParameterOverride = "GeneralReferral"
+    )
+    private List<ListValue<GeneralReferral>> generalReferrals;
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java:118]: are we displaying this on General Referral tab ?

Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java:118]: The tab update will be another PR

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralReferral.java (hunk 1 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralReferral.java]: Update in this commit https://github.com/hmcts/nfdiv-case-api/pull/1269/commits/e0baf9e5588269dde2f610f2464e6d1ed66ffe45

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**